### PR TITLE
add all, create, retrieve methods to Subscriptions, use v1/subs url

### PR DESCRIPTION
**Pull request by jxu577**
cc @kyleconroy @cupcait

### CODE REVIEW

**Comment by cupcait** [comment]
cc @stripe/api-libraries

File: src/main/java/com/stripe/model/SubscriptionCollection.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package com.stripe.model;
+
+public class SubscriptionCollection extends StripeCollection<Customer> {
```
Comment (kyleconroy) [inline src/main/java/com/stripe/model/SubscriptionCollection.java]: I believe this should be `StripeCollection<Subscription>`

File: src/main/java/com/stripe/model/Subscription.java [hunk 2]
```diff
@@ -63,7 +82,63 @@ public Subscription cancel(Map<String, Object> params, String apiKey) throws Aut
...
+
+	@Deprecated
+	public static SubscriptionCollection all(Map<String, Object> params,
+			String apiKey) throws AuthenticationException,
+			InvalidRequestException, APIConnectionException, CardException,
+			APIException {
+		return list(params, RequestOptions.builder().setApiKey(apiKey).build());
+	}
+
+	@Deprecated
```
Comment (cupcait) [inline src/main/java/com/stripe/model/Subscription.java]: Do we actually need to add new deprecated methods here?

Comment (kyleconroy) [inline src/main/java/com/stripe/model/Subscription.java]: That's a good point. Let's not add them.

File: src/main/java/com/stripe/model/SubscriptionCollection.java (hunk 1 above)
Comment (kyleconroy) [inline src/main/java/com/stripe/model/SubscriptionCollection.java]: I'm also a bit concerned that the tests still passed.

Comment (kyleconroy) [inline src/main/java/com/stripe/model/SubscriptionCollection.java]: Any idea why the tests didn't fail?

Comment (jxu577) [inline src/main/java/com/stripe/model/SubscriptionCollection.java]: hm I believe the Java tests just verify that the correct call to the API is made but does not actually execute the request and render the response

**Comment by kyleconroy** [comment]
We'll need to add `created` to the Subscription.java. We'll also need to rebase on master. Other than that, looks good to go.

**Pull request closed without merging**